tools/halide_image.h

```diff
@@ -223,9 +227,14 @@ class Image {
         contents->buf.min[2] = z;
         contents->buf.min[3] = w;
     }
+
+    void info(const char *tag = "Image") const;
+    void dump(const char *tag = "Image") const;
```

Comment:
I tried making these standalone and ran into another problem as these need to be templatized on the type of the Image class.